src/vcpkg/configuration.cpp

```diff
@@ -679,6 +688,71 @@ namespace vcpkg
 
     Json::IDeserializer<Configuration>& get_configuration_deserializer() { return ConfigurationDeserializer::instance; }
 
+    Optional<Configuration> parse_configuration(const Filesystem& fs, const Path& path, MessageSink& messageSink)
+    {
+        if (!fs.exists(path, IgnoreErrors{}))
+        {
+            return nullopt;
+        }
+
+        std::error_code ec;
+        auto conf = Json::parse_file(fs, path, ec);
+        if (ec)
+        {
+            messageSink.println(Color::error, msgFailedToRead, msg::path = path, msg::error_msg = ec);
+            return nullopt;
+        }
+        else if (!conf)
+        {
+            messageSink.println(msgFailedToParseConfig, msg::path = path);
+            messageSink.println(Color::error, LocalizedString::from_raw(conf.error()->to_string()));
+            return nullopt;
+        }
+
+        auto conf_value = std::move(conf.value_or_exit(VCPKG_LINE_INFO));
+        if (!conf_value.first.is_object())
+        {
+            messageSink.println(msgFailedToParseNoTopLevelObj, msg::path = path);
+            return nullopt;
+        }
+
+        return parse_configuration(std::move(conf_value.first.object(VCPKG_LINE_INFO)), path, messageSink);
+    }
+
+    Optional<Configuration> parse_configuration(const Json::Object& obj, StringView origin, MessageSink& messageSink)
+    {
+        Json::Reader reader;
+        auto maybe_configuration = reader.visit(obj, get_configuration_deserializer());
+        bool has_warnings = !reader.warnings().empty();
+        bool has_errors = !reader.errors().empty();
+        if (has_warnings || has_errors)
+        {
+            if (has_errors)
+            {
+                messageSink.println(Color::error, msgFailedToParseConfig, msg::path = origin);
+            }
+            else
+            {
+                messageSink.println(Color::warning, msgWarnOnParseConfig, msg::path = origin);
+            }
+
+            for (auto&& msg : reader.warnings())
+            {
+                auto warning = fmt::format("    {}\n", msg);
+                messageSink.println(Color::warning, LocalizedString::from_raw(warning));
```

Comment:
```suggestion
                messageSink.println(Color::warning, LocalizedString().append_indent().append_raw(msg));
```

Ditto below for error